python/cudf/cudf/core/column/column.py

```diff
@@ -1249,19 +1251,28 @@ def _all_bools_with_nulls(
     ) -> ColumnBase:
         # Might be able to remove if we share more of
         # DatetimeColumn._binaryop & TimedeltaColumn._binaryop
-        if self.has_nulls() and other.has_nulls():
+        if self.has_nulls() and (
```

Comment:
This if-else cascade is structured in a way that could produce multiple duplicate checks. What if we instead start with `result_mask=None`, then we branch on `self.has_nulls()` first, then do nested conditions for the rest? We should be able to structure this such that we never check a condition more than once, and if we initialize `result_mask` to `None` then it's fine if none of the conditions are met since we wind up back in the default case (the current else clause).